packages/jest/src/create-enzyme-serializer.js

```diff
@@ -3,8 +3,66 @@ import type { Options } from './create-serializer'
 import { createSerializer as createEmotionSerializer } from './create-serializer'
 import * as enzymeTickler from './enzyme-tickler'
 import { createSerializer as createEnzymeToJsonSerializer } from 'enzyme-to-json'
+import {
+  isEmotionCssPropElementType,
+  isStyledElementType,
+  unwrapFromPotentialFragment
+} from './utils'
 
-const enzymeSerializer = createEnzymeToJsonSerializer({})
+const enzymeToJsonSerializer = createEnzymeToJsonSerializer({
+  map: json => {
```

Comment:
Note for me, we probably should try to map this in our implementation - just how we do it for CSS prop elements. I've always got it working with this diff:
<details>
<summary>diff</summary>

```diff
diff --git a/packages/jest/src/create-enzyme-serializer.js b/packages/jest/src/create-enzyme-serializer.js
index 52ec73be..47787dbe 100644
--- a/packages/jest/src/create-enzyme-serializer.js
+++ b/packages/jest/src/create-enzyme-serializer.js
@@ -9,21 +9,7 @@ import {
   unwrapFromPotentialFragment
 } from './utils'
 
-const enzymeToJsonSerializer = createEnzymeToJsonSerializer({
-  map: json => {
-    if (typeof json.node.type === 'string') {
-      return json
-    }
-    const isRealStyled = json.node.type.__emotion_real === json.node.type
-    if (isRealStyled) {
-      return {
-        ...json,
-        children: json.children.slice(-1)
-      }
-    }
-    return json
-  }
-})
+const enzymeToJsonSerializer = createEnzymeToJsonSerializer()
 
 // this is a hack, leveraging the internal/implementation knowledge about the enzyme's ShallowWrapper
 // there is no sane way to get this information otherwise though
diff --git a/packages/jest/src/create-serializer.js b/packages/jest/src/create-serializer.js
index 75b984c3..a89b5cce 100644
--- a/packages/jest/src/create-serializer.js
+++ b/packages/jest/src/create-serializer.js
@@ -13,7 +13,9 @@ import {
   getKeys,
   flatMap,
   isPrimitive,
-  hasIntersection
+  hasIntersection,
+  isStyledElementType,
+  isStyledComponentType
 } from './utils'
 
 function getNodes(node, nodes = []) {
@@ -149,6 +151,12 @@ const createConvertEmotionElements = (keys: string[]) => (node: any) => {
       type: node.props.__EMOTION_TYPE_PLEASE_DO_NOT_USE__
     }
   }
+  if (isStyledComponentType(node.node)) {
+    return {
+      ...node,
+      children: node.children.slice(-1)
+    }
+  }
   if (isReactElement(node)) {
     return copyProps({}, node)
   }
diff --git a/packages/jest/src/utils.js b/packages/jest/src/utils.js
index 6eb7dbd1..72f655d1 100644
--- a/packages/jest/src/utils.js
+++ b/packages/jest/src/utils.js
@@ -98,12 +98,17 @@ export function isEmotionCssPropElementType(val: any): boolean {
   )
 }
 
+export function isStyledComponentType(val: any): boolean {
+  if (!val) return false
+  const { type } = val
+  return type.__emotion_real === type
+}
+
 export function isStyledElementType(val: any): boolean {
   if (val.$$typeof !== Symbol.for('react.element')) {
     return false
   }
-  const { type } = val
-  return type.__emotion_real === type
+  return isStyledComponentType(val)
 }
 
 export function isEmotionCssPropEnzymeElement(val: any): boolean {

```
</details>

The only thing missing here is that this should flatten multiple (two?) levels of our "wrappers". After unwrapping the CSS prop element we should also try to unwrap the styled element.

I decided that I don't care about this that much now though, the current solution is working fine so I prefer to land this sooner than later and maybe revisit this in the future.